akka-http-caching/src/main/java/akka/http/caching/javadsl/Cache.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com/>
+ */
+package akka.http.caching.javadsl;
+
+import akka.annotation.ApiMayChange;
+import akka.annotation.DoNotInherit;
+import akka.japi.Creator;
+import akka.japi.Procedure;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+@ApiMayChange
+@DoNotInherit
+public interface Cache<K, V> {
+  /**
+   * Returns either the cached CompletionStage for the given key or evaluates the given value generating
+   * function producing a `CompletionStage<V>`.
+   */
+  CompletionStage<V> getFuture(K key, Creator<CompletionStage<V>> genValue);
+
+  /**
+   * Returns either the cached {@code CompletionStage} for the key, or evaluates the given function which
+   * should lead to eventual completion of the completable future.
+   */
+  CompletionStage<V> getOrFulfil(K key, Procedure<CompletableFuture<V>> f);
+
+  /**
+   * Returns either the cached CompletionStage for the given key or the given value as a CompletionStage
+   */
+  CompletionStage<V> getOrCreateStrict(K key, Creator<V> block);
+
+  /**
+   * Retrieves the CompletionStage instance that is currently in the cache for the given key.
+   * Returns None if the key has no corresponding cache entry.
+   */
+  Optional<CompletionStage<V>> getOptional(K key);
```

Comment:
Caffeine uses `getIfPresent`, where a `get` is computing (as that is the preferred idiom). That is to say the behavior of the Cache is active (get/getIfPresent), whereas a Map is passive (get/computeIfAbsent). Also, `Optional` uses `ifPresent` which implies it leans towards being populated. I think trying to be consistent with the Java Collections Framework is a good API idiom.